tests/linearizability/client.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linearizability
+
+import (
+	"context"
+	"time"
+
+	"github.com/anishathalye/porcupine"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.uber.org/zap"
+)
+
+type recordingClient struct {
+	client   clientv3.Client
+	id       int
+	baseTime time.Time
```

Comment:
I don't think we need `baseTime`.  The [Operation.Call](https://github.com/anishathalye/porcupine/blob/7cbc8342681863b08e96432402d0588fc1defa88/model.go#L8) and [Operation.Return](https://github.com/anishathalye/porcupine/blob/7cbc8342681863b08e96432402d0588fc1defa88/model.go#L10) are just the timestamp when the operation is invoked and when it's returned, instead of duration.